include/picongpu/plugins/saxs/Saxs.hpp

```diff
@@ -0,0 +1,481 @@
+            isMaster = false; // no Master anymore -> no process is able to write
+        }
+        else
+        {
+            ofile <<  "Number of particles:"  << " " << np_master.x() << "\n";
+            ofile <<  "Number of macro particles:"  << " " << nmp_master << "\n";
+
+            /* if (ofile.fail())
+                std::cerr << "Error on flushing file [" << name << "]. " << std::endl; */
+
+            ofile.close();
+        }
+    }
+
+
+    void pluginUnload()
+    {
+        if(!notifyPeriod.empty())
+        {
+            __delete(sumfcoskr);
+            __delete(sumfsinkr);
+            __delete(np);
+            __delete(nmp);
+            __deleteArray(intensity_master);
+            CUDA_CHECK(cudaGetLastError());
+        }
+    }
+
+
+  /** Method to copy data from GPU to CPU */
+  void copyIntensityDeviceToHost()
+  {
+    sumfcoskr->deviceToHost();
+    sumfsinkr->deviceToHost();
+    np->deviceToHost();
+    nmp->deviceToHost();
+    __getTransactionEvent().waitForFinished();
+  }
+
+
+
+
+  /** combine radiation data from each CPU and store result on master
+   *  copyRadiationDeviceToHost() should be called before */
+  void collectIntensityOnMaster()
+  {
+      reduce(nvidia::functors::Add(),
+             sumfcoskr_master,
+             sumfcoskr->getHostBuffer().getBasePointer(),
+             n_q,
+             mpi::reduceMethods::Reduce()
+             );
+      reduce(nvidia::functors::Add(),
+             sumfsinkr_master,
+             sumfsinkr->getHostBuffer().getBasePointer(),
+             n_q,
+             mpi::reduceMethods::Reduce()
+             );
+      reduce(nvidia::functors::Add(),
+             &np_master,
+             np->getHostBuffer().getBasePointer(),
+             1,
+             mpi::reduceMethods::Reduce()
+             );
+      reduce(nvidia::functors::Add(),
+             &nmp_master,
+             nmp->getHostBuffer().getBasePointer(),
+             1,
+             mpi::reduceMethods::Reduce()
+             );
+        
+      // Calculate intensity on master
+      if (isMaster)
+      {
+        // printf("np_master = %f\n", np_master.x());
+        for ( unsigned int i = 0; i < n_q; i++)
+            intensity_master[i] = (sumfcoskr_master[i]*sumfcoskr_master[i]+sumfsinkr_master[i]*sumfsinkr_master[i])/np_master.x();
+        std::stringstream o_step;
+              o_step << currentStep;
+        writeQintensity(intensity_master, "saxsOutput/" + filename_prefix + "_" + o_step.str() + ".dat");
+        writeLog(np_master,nmp_master,"saxsOutput/" + filename_prefix + "_" + o_step.str() + ".log");
+      } 
+        
+  }
+
+
+
+
+  /** perform all operations to get data from GPU to master */
+  void collectDataGPUToMaster()
+  {
+      // collect data GPU -> CPU -> Master
+      copyIntensityDeviceToHost();
+      collectIntensityOnMaster();
+  }
+
+
+ 
+
+
+  /**
+   * This functions calls the radiation kernel. It specifies how the
```

Comment:
Please update this comment slightly to fit to the SAXS plugin:
Mainly: `radiation` -> `SAXS`